sql/importer: TestImportIntoRowCountCheckAfterPause failed

sql/importer.TestImportIntoRowCountCheckAfterPause [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21107688?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21107688?buildTab=artifacts#/) on master @ [becd8a914e851b2b9cc2c4021f5e28c75a94c133](https://github.com/cockroachdb/cockroach/commits/becd8a914e851b2b9cc2c4021f5e28c75a94c133):


```
=== RUN   TestImportIntoRowCountCheckAfterPause
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/d93b985d3dfb617d1a8f3c9c3f6a2f89/logTestImportIntoRowCountCheckAfterPause1524051769
    test_log_scope.go:82: use -show-logs to present logs inline
    import_into_test.go:517: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d93b985d3dfb617d1a8f3c9c3f6a2f89/logTestImportIntoRowCountCheckAfterPause1524051769
--- FAIL: TestImportIntoRowCountCheckAfterPause (2.96s)
=== RUN   TestImportIntoRowCountCheckAfterPause/empty_table
    import_into_test.go:488: error scanning '&{<nil> 0x4004948d20}': sql: no rows in result set
--- FAIL: TestImportIntoRowCountCheckAfterPause/empty_table (0.64s)
=== RUN   TestImportIntoRowCountCheckAfterPause/nonempty_table
    sql_runner.go:317: error executing 'SELECT count(*) FROM nonempty_table': pq: relation "nonempty_table" is offline: importing
--- FAIL: TestImportIntoRowCountCheckAfterPause/nonempty_table (1.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportIntoRowCountCheckAfterPause.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60259

--- a/pkg/sql/importer/import_into_test.go
+++ b/pkg/sql/importer/import_into_test.go
@@ -481,11 +481,15 @@ func TestImportIntoRowCountCheckAfterPause(t *testing.T) {
 			require.Error(t, err)
 			require.ErrorContains(t, err, "pause point")
 
-			// Find the paused import job.
+			// Find the paused import job. The job transitions from
+			// pause-requested to paused asynchronously, so retry until
+			// it reaches the paused state.
 			var jobID int64
-			runner.QueryRow(t,
-				`SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'IMPORT' AND status = 'paused'`,
-			).Scan(&jobID)
+			testutils.SucceedsSoon(t, func() error {
+				return db.QueryRow(
+					`SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'IMPORT' AND status = 'paused'`,
+				).Scan(&jobID)
+			})
 
 			// Clear pausepoint and resume the job.
 			runner.Exec(t, `SET CLUSTER SETTING jobs.debug.pausepoints = ''`)

